app/controllers/api/deploy_groups_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+class Api::DeployGroupsController < Api::BaseController
+  include DeployGroupsCommon
+  skip_before_action :require_project, only: :index
+  before_action :enabled?
+
+  def index
+    render json: paginate(deploy_groups)
+  end
+
+  def project_deploy_groups
+    render json: paginate(stage.deploy_groups)
+  end
+
+  def update_deploy_groups
+    if !production_change?
+      stage.deploy_groups = deploy_groups_to_add
+      stage.save!
+      head :no_content
+    else
+      render json: {message: "Do not modify production via the API"}, status: 403
```

Comment:
but it's an authorized request ... not sure why it matters how it came to
the app ... also some very arbitrary rule that makes the api useless ...

On Tue, Aug 16, 2016 at 1:10 PM, Adam Panzer <notifications@github.com>
wrote:

> In app/controllers/api/deploy_groups_controller.rb
> <https://github.com/zendesk/samson/pull/1198#discussion_r75010128>:
>
> > +
> > +  def index
> > +    render json: paginate(deploy_groups)
> > +  end
> > +
> > +  def project_deploy_groups
> > +    render json: paginate(stage.deploy_groups)
> > +  end
> > +
> > +  def update_deploy_groups
> > +    if !production_change?
> > +      stage.deploy_groups = deploy_groups_to_add
> > +      stage.save!
> > +      head :no_content
> > +    else
> > +      render json: {message: "Do not modify production via the API"}, status: 403
>
> This was because we don't want non-humans making production changes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1198/files/d43bdda94cc05d08fa093f7c694b09953335fe89#r75010128>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ3AvMCswYx1rNRuA2lzV4eQUV9dCks5qghkrgaJpZM4Jhtog>
> .
>